internal/backends/winit/renderer/femtovg.rs

```diff
@@ -349,12 +369,38 @@ impl super::WinitCompatibleCanvas for FemtoVGCanvas {
     }
 
     fn component_destroyed(&self, component: i_slint_core::component::ComponentRef) {
-        self.graphics_cache.component_destroyed(component)
+        self.opengl_context
+            .with_current_context(|_| self.graphics_cache.component_destroyed(component))
+    }
+
+    fn with_graphics_api(&self, callback: impl FnOnce(i_slint_core::api::GraphicsAPI<'_>)) {
+        #[cfg(not(target_arch = "wasm32"))]
+        let api = GraphicsAPI::NativeOpenGL {
+            get_proc_address: &|name| self.opengl_context.get_proc_address(name),
+        };
+        #[cfg(target_arch = "wasm32")]
+        let canvas_element_id = opengl_context.html_canvas_element().id();
+        #[cfg(target_arch = "wasm32")]
+        let api = GraphicsAPI::WebGL {
+            canvas_element_id: canvas_element_id.as_str(),
+            context_type: "webgl",
+        };
```

Comment:
I find the #[cfg...] so confusing to read!

Would it make sense to group the two wasm32 ones? E.g.

```rust
#[cfg(target_arch = "wasm32")]
let api = {
    let canvas_element_id = opengl_context.html_canvas_element().id();
    GraphicsAPI::WebGL {
        canvas_element_id: canvas_element_id.as_str(),
        context_type: "webgl",
    }
}
```

Reply:
Sure, can do that. There's more fixing on the wasm side I realised (well, the CI did), so I'll change that but also will force-push to avoid breaking the build in the earlier commits.

Reply:
Ahh, it doesn't work because `canvas_element_id` is borrowed and needs to live a long as `GraphicsAPI` :(

Reply:
I can make two versions of the function though